mailet/api/src/main/java/org/apache/mailet/MailetContext.java

```diff
@@ -123,18 +125,40 @@
      * retrieve an attribute's value given its name.
      *
      * @return an Iterator (of Strings) over all attribute names
+     * @deprecated see {@link #attributeNames()}
      */
+    @Deprecated
     Iterator<String> getAttributeNames();
 
+    /**
+     * Returns a Stream over the names of all attributes which are set
+     * in this Mail instance.
+     * <p>
+     * The {@link #getAttribute} method can be called to
+     * retrieve an attribute's value given its name.
+     *
+     * @since Mailet API v3.2
+     */
+    Stream<AttributeName> attributeNames();
```

Comment:
The `Iterator<String> getAttributeNames()` method is used nowhere.

Why not just deprecate it without replacement?